x-pack/plugins/security_solution/server/lib/detection_engine/signals/alert_suppression/group_and_bulk_create.ts

```diff
@@ -188,25 +199,33 @@ export const groupAndBulkCreate = async ({
         ruleExecutionLogger: runOpts.ruleExecutionLogger,
       });
 
-      const bulkCreateResult = await runOpts.bulkCreate(wrappedAlerts);
-
-      addToSearchAfterReturn({ current: toReturn, next: bulkCreateResult });
-
-      runOpts.ruleExecutionLogger.debug(`created ${bulkCreateResult.createdItemsCount} signals`);
-
-      const newBucketHistory: BucketHistory[] = buckets
-        .filter((bucket) => {
-          return !Object.values(bucket.key).includes(null);
-        })
-        .map((bucket) => {
-          return {
-            // This cast should be safe as we just filtered out buckets where any key has a null value.
-            key: bucket.key as Record<string, string | number>,
-            endDate: bucket.max_timestamp.value_as_string
-              ? bucket.max_timestamp.value_as_string
-              : tuple.to.toISOString(),
-          };
+      const suppressionDuration = runOpts.completeRule.ruleParams.alertSuppression?.duration;
+
+      if (suppressionDuration) {
+        const suppressionWindow = `now-${suppressionDuration.value}${suppressionDuration.unit}`;
+        const bulkCreateResult = await bulkCreateWithSuppression({
+          alertWithSuppression: runOpts.alertWithSuppression,
+          ruleExecutionLogger: runOpts.ruleExecutionLogger,
+          wrappedDocs: wrappedAlerts,
+          services,
+          suppressionWindow,
+          alertTimestampOverride: runOpts.alertTimestampOverride,
         });
+        addToSearchAfterReturn({ current: toReturn, next: bulkCreateResult });
+      } else {
+        const bulkCreateResult = await runOpts.bulkCreate(wrappedAlerts);
+        addToSearchAfterReturn({ current: toReturn, next: bulkCreateResult });
+        runOpts.ruleExecutionLogger.debug(`created ${bulkCreateResult.createdItemsCount} signals`);
```

Comment:
So I wanted to ask what our logging/debugging situation will look like with suppressed alerts -- i.e. when debugging via logs or the`.kibana-event-log*`, can we determine when an execution resulted in suppressed alerts, and any relevant counts (total alerts, suppressed alerts, etc).

Then I remembered we never got to merge https://github.com/elastic/kibana/pull/126210 which was to add `alertsDetected` and `alertsCreated` metrics for Security rules... 😔 

I think our plan right now is to wait on https://github.com/elastic/kibana/issues/130966 to be merged, then we can return these metrics back to the framework to write (in the pre-existing `kibana.alert.rule.execution.metrics.alert_counts` metric fields stack rules are writing now). Once that's complete, it would be nice to record any relevant execution metadata with regards to alert suppression.

That said, thankfully with this feature we only suppress if an alert already exists, and we update the alert itself with some of this information, so it won't all be lost and can be recovered from the alert if need be for debugging.